Sources/Instrumentation/Tracing/Span.swift

```diff
@@ -0,0 +1,224 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Tracing open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Swift Tracing project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Baggage
+import Dispatch
+
+/// A `Span` type that follows the OpenTracing/OpenTelemetry spec. The span itself should not be
+/// initializable via its public interface. `Span` creation should instead go through `tracer.startSpan`
+/// where `tracer` conforms to `TracingInstrument`.
+public protocol Span {
+    /// The operation name is a human-readable string which concisely identifies the work represented by the `Span`.
+    ///
+    /// For guideline on how to name `Span`s, please take a look at the
+    /// [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#span).
+    var operationName: String { get }
+
+    /// The status of this span.
+    var status: SpanStatus? { get set }
+
+    /// The precise `DispatchTime` of when the `Span` was started.
+    var startTimestamp: DispatchTime { get }
+
+    /// The precise `DispatchTime` of when the `Span` has ended.
+    var endTimestamp: DispatchTime? { get }
+
+    /// The read-only `BaggageContext` of this `Span`, set when starting this `Span`.
+    var baggage: BaggageContext { get }
+
+    /// The read-only collection of events which happened within this `Span`.
+    ///
+    /// `SpanEvent`s might be added via the `addEvent` or `addingEvent` `Span` methods.
+    var events: [SpanEvent] { get }
+
+    /// Add a `SpanEvent` in place.
+    /// - Parameter event: The `SpanEvent` to add to this `Span`.
+    mutating func addEvent(_ event: SpanEvent)
+
+    // TODO: Wrap in a struct to hide collection implementation details.
```

Comment:
ticketify or do right away please, your call :)

Reply:
I'll do the wrapping in this PR, then we can think about implementation details in an upcoming issue.